packages/cli/src/commands/help.ts

```diff
@@ -0,0 +1,212 @@
+    const optionLine = optionLines[i];
+    const option = command.options[i];
+    // Add only 2 spaces to the longest line, and then make all shorter lines the same length.
+    optionLine.push(
+      ' '.repeat(2 + (maxLineStartLength - calcLineLength(optionLine)))
+    );
+
+    // Descriptions may be longer than max line length. Wrap them to the same column as the first description line
+    const lines: string[][] = [optionLine];
+    for (const descriptionWord of option.description.split(' ')) {
+      // insert a new line when the next word would match or exceed the maximum line length
+      if (
+        calcLineLength(lines[lines.length - 1]) +
+          stripAnsi(descriptionWord).length >=
+        MAX_LINE_LENGTH
+      ) {
+        // initialize the new line with the necessary whitespace. The INDENT is apart of `maxLineStartLength`
+        lines.push([' '.repeat(maxLineStartLength + 2)]);
+      }
+      // insert the word to the current last line
+      lines[lines.length - 1].push(descriptionWord);
+    }
+    // for every line, transform into a string and push it to the output
+    for (const line of lines) {
+      outputArray.push(lineToString(line));
+    }
+    // add an empty line in between in each option block for readability (skip the last block)
+    if (i !== optionLines.length - 1) outputArray.push('');
+  }
+
+  // return the entire list of options as a single string after delete the last '\n' added to the option list
+  const outputString = outputArrayToString(outputArray);
+  return outputString.substring(0, outputString.length - 1);
+}
+
+function buildCommandExampleLines(command: Command) {
+  const outputArray: string[] = [chalk.dim(`Examples:`), ''];
+  for (const example of command.examples) {
+    const nameLine: string[] = [INDENT];
+    nameLine.push(chalk.gray('-'));
+    nameLine.push(example.name);
+    outputArray.push(lineToString(nameLine));
+    outputArray.push('');
+    const buildValueLine = (value: string) => {
+      return lineToString([INDENT, INDENT, chalk.cyan(`$ ${value}`)]);
+    };
+    if (Array.isArray(example.value)) {
+      for (const line of example.value) {
+        outputArray.push(buildValueLine(line));
+      }
+    } else {
+      outputArray.push(buildValueLine(example.value));
+    }
+    outputArray.push('');
+  }
+  // delete the last newline added after examples iteration
+  outputArray.splice(-1);
+  // delete the last newline appended to the last example line
+  const outputString = outputArrayToString(outputArray);
+  return outputString.substring(0, outputString.length - 1);
+}
+
+function buildHelpOutput(command: Command) {
+  const outputArray: string[] = [
+    buildCommandSynopsisLine(command),
+    '',
+    command.description,
+    '',
+    buildCommandOptionLines(command),
+    '',
+    buildCommandExampleLines(command),
+  ];
+
+  return outputArrayToString(outputArray);
+}
+
+export function help(command: Command) {
+  return buildHelpOutput(command);
+}
```

Comment:
<!-- Suggestion comments propose improvements to the code. Mention specifically what should be changed and what the benefit would be.  -->
**suggestion (blocking):** ‏add tests

This logic looks like it's doing what was intended, but it would be really easy to be off by one somewhere. Can we get some tests around this logic?

Reply:
added!